net_sink.go

```diff
@@ -275,7 +275,7 @@ func (s *netSink) run() {
 	for {
 		if s.conn == nil {
 			if err := s.connect(addr); err != nil {
-				s.log.Warnf("statsd connection error: %s", err)
+				s.log.Warn(fmt.Errorf("gostats connection error: %w", err))
```

Comment:
nit: we could just call it a connection error. If the caller wants the log to be prefixed, they can provide a named or pre-prefixed logger to the netSink set-up.